docs/main/04-guides/04-writing-a-promise.mdx

```diff
@@ -0,0 +1,259 @@
+and we will need to provision a Postgres database in this workflow.
+
+To start, we can use the Kratix CLI again to add a container to the Resource Configure workflow:
+```
+kratix add container resource/configure/database-configure --image kratix-guide/database-resource-pipeline:v0.1.0 --language python
+```
+
+Take a look at the file `workflows/resource/configure/database-configure/kratix-guide-database-resource-pipeline/scripts/pipeline.py`
+
+This is our Resource Configure script, and it's written in Python using the [Kratix Python SDK](/main/kratix-sdks/python).
+It has some print lines and let's replace the content, so it generates a Postgres database for us:
+
+```python
+import kratix_sdk as ks
+import yaml
+
+def main():
+    sdk = ks.KratixSDK()
+    resource = sdk.read_resource_input()
+    name = resource.get_name()
+    size = resource.get_value("spec.size")
+
+    manifest = {
+        "apiVersion": "acid.zalan.do/v1",
+        "kind": "postgresql",
+        "metadata": {"name": name, "namespace": "default"},
+        "spec": {
+        	"teamId": "kratix",
+			"enableLogicalBackup": True,
+        	"volume": {
+            	"size": size
+        	},
+        	"numberOfInstances": 2,
+                "users": {
+                     "team-a": ["superuser", "createdb"]
+                },
+        	"postgresql": {
+            	"version": "16"
+        	}
+		}
+    }
+    data = yaml.safe_dump(manifest).encode("utf-8")
+    sdk.write_output("database.yaml", data)
+
+    status = ks.Status()
+    status.set("teamId", "kratix")
+    sdk.write_status(status)
+
+
+if __name__ == '__main__':
+    main()
+```
+
+Let's build and load this image into our local environment:
+
+```bash
+docker build -t kratix-guide/database-resource-pipeline:v0.1.0 workflows/resource/configure/database-configure/kratix-guide-database-resource-pipeline
+kind load docker-image kratix-guide/database-resource-pipeline:v0.1.0 --name platform
+```
+
+:::info
+Always remember to load new images into your environment when using kind!
+:::
+
+### Using the Promise
+
+We are ready to install our database Promise! Install the Promise by running:
+```bash
+kubectl apply -f promise.yaml --context $PLATFORM
+```
+
+You can check on the installation of the Promise by running:
+
+```bash
+kubectl get promise database --context $PLATFORM -w
+
+NAME       STATUS        KIND       API VERSION               VERSION
+database   Unavailable   Database   test.kratix.io/v1alpha1   v0.0.1
+database   Available     Database   test.kratix.io/v1alpha1   v0.0.1
+```
+
+Once this Promise becomes available, we can make a resource request.
+Take a look at `example-resource.yaml` and set a size for this Postgres database:
+
+```yaml
+apiVersion: test.kratix.io/v1alpha1
+kind: Database
+metadata:
+  name: example-database
+spec:
+  size: "1Gi"
+```
+
+You can now make a request:
+```bash
+kubectl apply -f example-resource.yaml --context $PLATFORM
+```
+
+Check the status of this resource:
```

Comment:
Just to keep the narrative flow, maybe you could say: "Now that the request is made, let's check the status of this resource:"